tf2_geometry_msgs/CMakeLists.txt

```diff
@@ -42,4 +42,21 @@ if(BUILD_TESTING)
   endif()
 endif()
 
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME} INTERFACE $<INSTALL_INTERFACE:include>)
+ament_target_dependencies(${PROJECT_NAME} INTERFACE ${required_dependencies})
+
+install(
+  TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}
+  INCLUDES DESTINATION include
+)
+
+ament_export_include_directories(include)
+ament_export_targets(${PROJECT_NAME})
+ament_export_dependencies(${required_dependencies})
+
 ament_auto_package()
+
+install(
+  DIRECTORY include/
+  DESTINATION include)
```

Comment:
If we're going to do all of this by hand, then we should get rid of `ament_auto_package`, since it also does a lot of the same (though with some different flags).

Reply:
I would keep that to a follow up PR in order to get this important bug fix merged asap.

Reply:
I don't really understand why, to be honest.  We are changing how this works, and we have to test the downstreams of `tf2_geometry_msgs` anyway with this change.  So we may as well make it how it should be.

Reply:
Two reasons:

* Refactoring the CMake logic of this package is completely unrelated to the issue this patch fixes. For that reason alone it should be done in a separate PR.
* Atm we have a severe issue - that packages can't be built on to of a fast archive. That needs to be fixed asap - especially since we want to freeze the rosdistro for Foxy today. And since there is still work which needs to be done to get there it is much more important than refactoring existing code for no functional benefit.

Reply:
>     * Refactoring the CMake logic of this package is completely unrelated to the issue this patch fixes. For that reason alone it should be done in a separate PR.

But that is what this patch is doing.  So that's why I don't understand.

>     * Atm we have a severe issue - that packages can't be built on top of a relocated install space (which is the case for fat archives). That needs to be fixed asap - especially since we want to freeze the rosdistro for Foxy today. And since there is still work which needs to be done to get there it is much more important than refactoring existing code for no functional benefit.

I thought we had decided to punt this to Patch Release 1.  Regardless, this isn't the right thing to do 2 days before a release.  We should either think about slipping the release, or slip this fix to Patch release 1.

Reply:
eb64bb5f9a231cf9dabc78231e0e16fcd9daa610 removes the redundant explicit call. After that the patch only adds missing exported dependencies as well as modern CMake targets.